core/src/main/scala/org/apache/spark/sql/hive/SnappyStoreHiveCatalog.scala

```diff
@@ -117,7 +116,9 @@ class SnappyStoreHiveCatalog(externalCatalog: SnappyExternalCatalog,
       // In smart connector, property name is different.
       user = snappySession.conf.get(Constant.SPARK_STORE_PREFIX + Attribute.USERNAME_ATTR, "")
     }
-    val defaultName = if (user.isEmpty) Constant.DEFAULT_SCHEMA else formatDatabaseName(user)
+    val defaultName = IdUtil.getUserAuthorizationId(
+      if (user.isEmpty) Constant.DEFAULT_SCHEMA else formatDatabaseName(user)).
+      replace('-', '_')
```

Comment:
replace can be done for only else part

Reply:
That is true, but since DEFAULT_SCHEMA though right now app, is hypothetically is changed to include - , then it can create trouble. Code is meant to be that at the end of all processing of name, if '-' remains , it is to be changed to '_'

Reply:
@sumwale I did not get the question.  are you talking about the situation where you have two users with name differing only by  - & _ . Hmm. Yes that will lead to clash... . This is a temporary measure, I suppose. I looked at changing the Hive related class to allow special character but decided against it as it would have required us to maintain another 3rd party repo branch internally. Once this JnJ is out of way, can work for a better fix as suggested.